pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Producer.java

```diff
@@ -244,6 +245,25 @@ public boolean checkAndStartPublish(long producerId, long sequenceId, ByteBuf he
         return true;
     }
 
+    private boolean checkCanProduceTxnOnTopic(long sequenceId, ByteBuf headersAndPayload) {
+        if (cnx.getBrokerService().pulsar().getConfiguration().isBlockTransactionsIfReplicationEnabled()) {
```

Comment:
we save us from this long chain of method calls by caching the value in the constructor
cnx.getBrokerService().pulsar().getConfiguration().isBlockTransactionsIfReplicationEnabled()